db/fns.md

```diff
@@ -2046,6 +2088,10 @@ client side.
 Update a queue artifact, including its storageType.
 Returns the up-to-date artifact row that have the same task id, run id, and name.
 
+### deprecated methods
+
+* `get_task_group(task_group_id_in text)` (compatibility guaranteed until v49.0.0)
```

Comment:
I don't think the database schema is a defined API into our system - the only approved route to collect data is via API calls, so I think we are free to change the database function inputs and outputs. I guess this is why we now have get_task_group2 - but I don't think it is needed.

Reply:
This is a convention. If one function is deprecated, we create new one with new name, and mark previous one as deprecated. 
We probably don't delete them after either (unless db:migrate command takes care of deprecated 2+ versions ago functions).

This is true that db is not a public api, but we have to keep it because of the running processes.  When we make a new deployment in kubernetes, there would be different versions of services running at the same time. Some background tasks might take hours to complete. So if migration removes those "unused" ones, it will quite likely break those processes in the middle of some work.
We can't change existing functions for the same reason.

Reply:
Ah, of course! Thanks Yarik!